jolt-core/src/jolt/instruction/virtual_advice.rs

```diff
@@ -0,0 +1,72 @@
+use ark_std::log2;
+use rand::prelude::StdRng;
+use rand::RngCore;
+use serde::{Deserialize, Serialize};
+
+use super::{JoltInstruction, SubtableIndices};
+use crate::jolt::subtable::{identity::IdentitySubtable, LassoSubtable};
+use crate::poly::field::JoltField;
+use crate::utils::instruction_utils::{chunk_operand_usize, concatenate_lookups};
+
+#[derive(Copy, Clone, Default, Debug, Serialize, Deserialize)]
+pub struct ADVICEInstruction<const WORD_SIZE: usize>(pub u64);
+
+impl<const WORD_SIZE: usize> JoltInstruction for ADVICEInstruction<WORD_SIZE> {
+    fn operands(&self) -> (u64, u64) {
+        (self.0, 0)
+    }
+
+    fn combine_lookups<F: JoltField>(&self, vals: &[F], C: usize, M: usize) -> F {
+        concatenate_lookups(vals, C, log2(M) as usize)
+    }
+
+    fn g_poly_degree(&self, _: usize) -> usize {
+        1
+    }
+
+    fn subtables<F: JoltField>(
+        &self,
+        C: usize,
+        _: usize,
+    ) -> Vec<(Box<dyn LassoSubtable<F>>, SubtableIndices)> {
+        vec![(
+            Box::new(IdentitySubtable::new()),
+            SubtableIndices::from(0..C),
+        )]
```

Comment:
this is pretty subtle, but for the ADVICE instruction we need to range-check that prover-supplied value is in `[0, 2^WORD_SIZE)`
By using `C` IdentitySubtables here, we're checking that the value is in the range `[0, M^C)` instead. For RV32 we use M=2^16 and C=4, so this range would be `[0, 2^64)` instead of `[0, 2^32)`. 

Instead, we can do what [`ADDInstruction`](https://github.com/a16z/jolt/blob/main/jolt-core/src/jolt/instruction/add.rs#L33-L49) does, using a combination of TruncateOverflowSubtable and IdentitySubtable to ensure that the output is at most `WORD_SIZE` bits